client/shipping-label/views/purchase/steps/packages/add-item.js

```diff
@@ -81,7 +80,16 @@ const AddItemDialog = ( {
 					{ itemOptions }
 				</div>
 				<ActionButtons buttons={ [
-					{ label: __( 'Add' ), isPrimary: true, onClick: () => moveItem( sourcePackageId, movedItemIndex, openedPackageId ) },
+					{ label: __( 'Add' ), isPrimary: true, onClick: () => {
+						Object.keys( addedItems || {} ).forEach( ( sourcePackageId ) => {
+							Object.keys( addedItems[ sourcePackageId ] || {} ).forEach( ( movedItemIndex ) => {
+								if ( addedItems[ sourcePackageId ][ movedItemIndex ] ) {
+									moveItem( sourcePackageId, movedItemIndex, openedPackageId );
```

Comment:
Why I must say, I think it would be a swell thing to refactor as it looks like the only way to fix my apparently broken original implementation 😄 that is, if multiple items were being moved from a single package, the indexing would be thrown off when acting in sequence.